feat(core): DebugNode.triggerEventHandler should accept eventObj as an optional parameter

JiaLiPassion opened the pull request:
Close https://github.com/angular/angular/issues/44724

`DebugNode.triggerEventHandler()` should accept the `eventObj` as an
optional parameter. So the user don't have to write code like

```
elem.triggerEventHandler('click', null);
```

mary-poppins commented:
You can preview 56152ea at https://pr45279-56152ea.ngbuilds.io/.

mary-poppins commented:
You can preview ad541e0 at https://pr45279-ad541e0.ngbuilds.io/.

mary-poppins commented:
You can preview 028006d at https://pr45279-028006d.ngbuilds.io/.

crisbeto commented on aio/content/examples/testing/src/app/dashboard/dashboard-hero.component.spec.ts:
```diff
@@ -1,14 +1,14 @@
-import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
-import { By } from '@angular/platform-browser';
-import { first } from 'rxjs/operators';
+import {DebugElement} from '@angular/core';
```
Should these formatting changes be reverted? I think that AIO follows different formatting rules than the rest of the repo.

JiaLiPassion replied:
@crisbeto , thank you for the review, I reverted the format.

JiaLiPassion committed:
feat(core): triggerEventHandler accept optional eventObj

Close #44724

`DebugNode.triggerEventHandler()` should accept the `eventObj` as an
optional parameter. So the user don't have to write code like

```
elem.triggerEventHandler('click', null);
```

JiaLiPassion committed:
docs(core): update triggerEventHandler samples

mary-poppins commented:
You can preview 2f857ee at https://pr45279-2f857ee.ngbuilds.io/.

crisbeto reviewed (approved):
LGTM

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

atscott reviewed (approved):
Reviewed-for: public-api

dylhunn reviewed (approved):
reviewed-for: public-api

AndrewKushnir commented:
[Presubmit](http://test/OCL:433383655:BASE:433383667:1646801819745:d1d24f8a).

atscott commented:
This PR was merged into the repository by commit c7cab69af1ef3ff6bf81b8075801031d16914e6f.

peterblazejewicz commented:
thx!
